index.js

```diff
@@ -116,9 +115,23 @@ const fastifyAutoload = async function autoload (fastify, options) {
 }
 
 async function getPackageType (cwd) {
-  const nearestPackage = await pkgUp({ cwd })
-  if (nearestPackage) {
-    return require(nearestPackage).type
+  const directories = cwd.split(path.sep)
+
+  // required for paths that begin with the sep, such as linux root
+  directories[0] = directories[0] !== '' ? directories[0] : path.sep
+
+  while (directories.length > 0) {
+    const file = path.join(...directories, 'package.json')
+
+    const fileExists = await fs.stat(file)
+      .then(() => true)
+      .catch(() => false)
+
+    if (fileExists) {
+      return require(file).type
```

Comment:
Please use `fs.readFile` instead

Reply:
Is there any particular reason for this? I only used `require` because the previous dependencies did, but am curious as it seems fairly fool-proof.

Cheers!